src/Hosting/Abstractions/src/BackgroundService.cs

```diff
@@ -13,7 +13,7 @@ namespace Microsoft.Extensions.Hosting
     public abstract class BackgroundService : IHostedService, IDisposable
     {
         private Task _executingTask;
-        private readonly CancellationTokenSource _stoppingCts = new CancellationTokenSource();
+        private CancellationTokenSource _stoppingCts;
```

Comment:
Dispose now has to handle the possibility that StartAsync was never called to initialize _stoppingCts. It should be updated to `_stoppingCts?.Cancel();`.